web/src/stream_topic_history.ts

```diff
@@ -103,26 +126,26 @@ export class PerStreamHistory {
           deleted.
     */
 
-    topics = new FoldDict();
+    topics = new FoldDict<TopicHistoryEntry>();
     // Most recent message ID for the stream.
     max_message_id = 0;
+    stream_id: number;
 
-    constructor(stream_id) {
+    constructor(stream_id: number) {
         this.stream_id = stream_id;
     }
 
-    has_topics() {
+    has_topics(): boolean {
         return this.topics.size !== 0;
     }
 
-    update_stream_max_message_id(message_id) {
+    update_stream_max_message_id(message_id: number): void {
         if (message_id > this.max_message_id) {
             this.max_message_id = message_id;
         }
     }
 
-    add_or_update(topic_name, message_id) {
-        message_id = Number.parseInt(message_id, 10);
+    add_or_update(topic_name: string, message_id: number): void {
```

Comment:
Why is is correct to remove the `parseInt` here?

Reply:
`add_or_update` is only called from `add_message`, which is called from three places

(1) `message_events.update_messages`, populated from `message_store.get`, so that's definitely a number.

(2) `message_helper.process_new_message` -- I tried adding `console.log(message.id, typeof message.id);` before the call to `add_message` and it showed me only numbers

(3) `echo.edit_locally` -- I checked all the callsites of `edit_locally` and they use either `message_lists.current.get(message_id)` (which isn't typed yet but I'm pretty sure returns a `Message` object) or `message_store.get`.

Given that, it seems like `message_id` is definitely a number and we don't need to `parseInt`.

Does that seem right?

Reply:
The `parseInt` was added in 5a8bccfe08ffe1a79144c88e50f64cf1abb5a019; maybe that was working around a bug where local-echo message IDs could be incorrectly a string.